### ISSUE TITLE
Is is possilble build an ios client?

### ISSUE THREAD

**Issue by orsfey** [SoftEtherVPN/SoftEtherVPN#545]
Is is possilble build an iso client like windows softether vpnclient?

**Comment by chipitsine** [comment]
yes, it is possible

**Comment by chipitsine** [comment]
@Gentoli , are you going to share your iOS code with upstream?

**Comment by Gentoli** [comment]
Yes, if I can get it working. But I don't know if my modifications are fitted for upstream.

**Comment by orsfey** [comment]
@davidebeatrici @Gentoli then do you have a plan to write an ios client? or looking for some ios developers to finish coding it?

**Comment by Gentoli** [comment]
@chipitsine 
Is there a way to make the client connect to the server is TUN mode so the server can handle things like DHCP?

**Comment by dvscomp2013** [comment]
any news on a ios client?

**Comment by evgenmedia** [comment]
Waiting for iOS SoftEther SDK 2 years ... It is really need for world community

**Comment by chipitsine** [comment]
@evgenmedia , it is up to you - wait for years or do something

**Comment by orsfey** [comment]
Waiting for iOS SoftEther SDK 4 years ...

**Comment by orsfey** [comment]
can you pls mark this post top? i hope some one who can write the app when see the post @chipitsine

**Comment by chipitsine** [comment]
I think if there were people willing to contribute, they would not have look at top post

**Comment by orsfey** [comment]
maybe, really home some one can write an ios vpnclient, if so, i can authentication vpn connection with certificate and custom port

**Comment by ricksuzade-maker** [comment]
If there is an iOS client, I would buy it ASAP.

**Comment by Evengard** [comment]
Well, I implemented certificate authentication via PPP-based protocols (L2TP, SSTP) aka EAP-TLS, my next step is to implement IPv6 support for them. I wonder if that wouldn't be enough...

**Comment by sagar-chandarana** [comment]
I wonder if this is related to SoftEther Protocol working on iOS, or a wrapper for existing protocols like L2TP.

**Comment by davidebeatrici** [comment]
As soon as support for WireGuard is implemented, you can use the official client on iOS (it's available on App Store).

**Comment by domosekai** [comment]
I made an client for iOS. It's called SSTP Connect. It's not free though.

**Comment by evgenmedia** [comment]
@domosekai can we buy your source code? If yes, please, contact with me evgen@dline-media.com

**Comment by igornazarov1991** [comment]
I have an issue of connecting on iOS using manual configuration when a protocol is IPsec. At the same time L2TP connection works fine. Does anyone experiencing the  same issues?
